spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/DecisionTreeRegressor.java

```diff
@@ -0,0 +1,243 @@
+    return getOutputSchema(inputSchema, config.predictionField);
+  }
+
+  private Schema getOutputSchema(Schema inputSchema, String predictionField) {
+    List<Schema.Field> fields = new ArrayList<>(inputSchema.getFields());
+    fields.add(Schema.Field.of(predictionField, Schema.of(Schema.Type.DOUBLE)));
+    return Schema.recordOf(inputSchema.getRecordName() + ".predicted", fields);
+  }
+
+  @Path("outputSchema")
+  public Schema getOutputSchema(GetSchemaRequest request) {
+    return getOutputSchema(request.inputSchema, request.predictionField);
+  }
+
+  /**
+   * Configuration for the DecisionTreeRegressor.
+   */
+  public static class DecisionTreeRegressorConfig extends PluginConfig {
+
+    @Description("The name of the FileSet to load the model from.")
+    private final String fileSetName;
+
+    @Description("Path of the FileSet to load the model from.")
+    private final String path;
+
+    @Nullable
+    @Description("A comma-separated sequence of fields to be used for Decision Tree Regression. Features to be used, " +
+      "must be from one of the following type: int, long, float or double. Both featuresToInclude and " +
+      "featuresToExclude fields cannot be specified.")
+    private final String featuresToInclude;
+
+    @Nullable
+    @Description("A comma-separated sequence of fields to be excluded when calculating prediction. Both " +
+      "featuresToInclude and featuresToExclude fields cannot be specified.")
+    private final String featuresToExclude;
+
+    @Description("The field on which to set the prediction. It will be of type double.")
+    private final String predictionField;
+
+    public DecisionTreeRegressorConfig(String fileSetName, String path, String featuresToInclude,
+                                       String featuresToExclude, String predictionField) {
+      this.fileSetName = fileSetName;
+      this.path = path;
+      this.featuresToInclude = featuresToInclude;
+      this.featuresToExclude = featuresToExclude;
+      this.predictionField = predictionField;
+    }
+
+    private void validate(Schema inputSchema) {
+      if (!Strings.isNullOrEmpty(featuresToExclude) && !Strings.isNullOrEmpty(featuresToInclude)) {
+        throw new IllegalArgumentException("Cannot specify values for both featuresToInclude and featuresToExclude. " +
+                                             "Please specify fields for one");
+      }
+      List<String> fields = getFeatureList(inputSchema);
+      for (String field : fields) {
+        Schema schema = inputSchema.getField(field).getSchema();
+        Schema.Type features = schema.isNullableSimple() ? schema.getNonNullable().getType() : schema.getType();
+        Preconditions.checkArgument(features.isSimpleType(), "Field to be used must be of type number: int, long, " +
+          "float, double but was of type %s for field %s ", features, field);
+        if (features.equals(Schema.Type.BYTES) || features.equals(Schema.Type.STRING) ||
```

Comment:
a little easier to read if you just do one check here instead of the isSimpleType() check and then this check.